### ISSUE TITLE
Unable to create FreeBSD helper for fstab mount

### ISSUE THREAD

**Issue by JRGonz** [rclone/rclone#7432]
-->

#### What is the problem you are having with rclone?

In linux, per rclone mount documentation, you simply have to symlink rclone to mount.rclone in order to get it to work with fstab. In FreeBSD the helpers are named mount_helper (in this case, rclone it should be mount_rclone). Unfortunately this does not allow the helper to be called from fstab since it does not detect that it should be in mount helper mode. 

rclone mount <remote> <local> works fine. 

It seems that possibly mount.rclone and rclonefs are explicitly set / hard coded but mount_rclone is not (for FreeBSD)? 

#### What is your rclone version (output from `rclone version`)

rclone v1.64.0-DEV

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)

freebsd 13.2-release-p4 (64 bit)

#### Which cloud storage system are you using? (e.g. Google Drive)

smb

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)

mount <mountpoint> (derived from /etc/fstab)

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)

Unfortunately I do not believe there is really anything to log here since FreeBSD does not use the mount.rclone helper and the mount_rclone helper does not work as a helper it just doesn't mount (using fstab). 

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by kapitainsky** [comment]
> It seems that possibly mount.rclone and rclonefs are explicitly set / hard coded but mount_rclone is not (for FreeBSD)?

You might be right:

https://github.com/rclone/rclone/blob/47ca0c326e8a84e47fa2988224356183a4686c8f/fs/mount_helper.go#L48-L57

It should be easy fix. As you are on FreeBSD you could try to rectify/test it and submit PR.

**Comment by JRGonz** [comment]
> It should be easy fix. As you are on FreeBSD you could try to rectify/test it and submit PR.
> 
> `if runtime.GOOS == "freebsd"` clause should solve the problem.

When I have time. I would need to install all those pesky go dev tools on something first. The fix looks straightforward from the context clues even though I'm not a developer. 

`if runtime.GOOS == "freebsd" {
 return me == "mount_rclone"
}`

or something of that nature?

**Comment by kapitainsky** [comment]
> When I have time. I would need to install all those pesky go dev tools on something first.

Only what you need is `go` and `git`

```
sudo pkg update && sudo pkg upgrade
sudo pkg install go
sudo pkg install git
```

> `if runtime.GOOS == "freebsd" { return me == "mount_rclone" }`
> 
> or something of that nature?

yea - something like that..

**Comment by JRGonz** [comment]
> > When I have time. I would need to install all those pesky go dev tools on something first.
> 
> Only what you need is `go` and `git`
> 
> ```
> sudo pkg update && sudo pkg upgrade
> sudo pkg install go
> sudo pkg install git
> ```

I came across a bug when looking for a solution to this problem that said cmount needed to be built into the fbsd rclone. Maybe I'm wrong. I'll try it like this in a little while and see what happens. Maybe it is only a FreeBSD 14 issue? 

[https://github.com/rclone/rclone/issues/5843](url)

**Comment by kapitainsky** [comment]
Based on this comment:

https://github.com/rclone/rclone/issues/5843#issuecomment-1784149722

I would stick to v1.65 with your go on this issue.

**Comment by JRGonz** [comment]
> Based on this comment:
> 
> [#5843 (comment)](https://github.com/rclone/rclone/issues/5843#issuecomment-1784149722)
> 
> I would stick to v1.65 with your go for this issue.

FreeBSD pkg repo has v1.20.8 :)  let me see what happens with that first.

**Comment by JRGonz** [comment]
fbsd% go build -tags cmount
# github.com/winfsp/cgofuse/fuse
../../go/pkg/mod/github.com/winfsp/cgofuse@v1.5.1-0.20221118130120-84c0898ad2e0/fuse/host_cgo.go:117:10: fatal error: 'fuse.h' file not found
#include <fuse.h>
         ^~~~~~~~
1 error generated.

and there don't seem to be any devel pkgs in the FreeBSD pkg repo for fuse :|

**Comment by kapitainsky** [comment]
You have to:

```
sudo pkg install gmake
gmake GOTAGS=cmount
```

binary will be created in `~/go/bin folder`

```
$ ~/go/bin/rclone --version
rclone v1.65.0-beta.7507.64ec5709f
- os/version: freebsd 13.2-release-p4 (64 bit)
- os/kernel: 13.2-release-p4 (amd64)
- os/type: freebsd
- os/arch: amd64
- go/version: go1.20.8
- go/linking: dynamic
- go/tags: cmount

```

**Comment by JRGonz** [comment]
I'm still running into build erros where it is telling me that the fuse header is unavailable even though it is  in 
/usr/local/include/fuse3/fuse.h 

fbsd% gmake GOTAGS=cmount
go build -v --ldflags "-s -X github.com/rclone/rclone/fs.Version=v1.65.0-beta.7507.64ec5709f" -tags "cmount"
github.com/winfsp/cgofuse/fuse
# github.com/winfsp/cgofuse/fuse
../../go/pkg/mod/github.com/winfsp/cgofuse@v1.5.1-0.20221118130120-84c0898ad2e0/fuse/host_cgo.go:117:10: fatal error: 'fuse.h' file not found
#include <fuse.h>
         ^~~~~~~~
1 error generated.
gmake: *** [Makefile:53: rclone] Error 1

**Comment by kapitainsky** [comment]
Are you on FreeBSD 14?

I will try it when it is released - should be any day now.

**Comment by JRGonz** [comment]
> fatal error: 'fuse.h' file not found
> #include <fuse.h>

No. 13.2. I can get it to build but when symlinking rclone_mount it says that mount is not available. I'm guessing because it needs to be built with cmount but there doesn't seem to be a fuse3 dev pkg to build against. There seems to be include files in /usr/local/include/fuse3 but gmake doesn't seem to think fuse.h is there. Think I'll just have to give up on this for a bit. It seemed like a pretty easy fix but I guess building it is another story.

**Comment by vedadkajtaz** [comment]
Until that's fixed, the following workaround works on FreeBSD:

* symlink `rclone` to `mount.rclone` in `/usr/local/bin`, which is a name `IsMountHelper()` expects
* in `fstab`, specify `fuse` instead of `rclone` as `FSType`, and pass the `mountprog` option

eg.

```
my:bucket /mnt/mybucket fuse rw,mountprog=/usr/local/bin/mount.rclone,config=...
```

**Comment by atagtm** [comment]
> Until that's fixed, the following workaround works on FreeBSD:
> 
There's nothing to be fixed, let's see what [mount(8)](https://man.freebsd.org/cgi/man.cgi?mount(8)) says:

> However,	for the	following  file	 system	 types:	 cd9660,  mfs,
	       msdosfs,	 nfs,  nullfs,	smbfs, udf, and	unionfs	mount will not
	       call [nmount(2)](https://man.freebsd.org/cgi/man.cgi?query=nmount&sektion=2&apropos=0&manpath=FreeBSD+14.2-RELEASE+and+Ports) directly and will	instead	attempt	to  execute  a
	       program	in /sbin/mount_type where type is replaced by the file
	       system type name.  For example, nfs file	systems	are mounted by
	       the program /sbin/mount_nfs.

This is exclusive only for those fs types. Have to add mountprog=.. to the options on FreeBSD, there's nothing to be fixed in the code.
